src/components/UI/DownloadButton/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import protectedAxios from 'utils/protectedAxios';
+import { AxiosResponse } from 'axios';
+import Loading from 'components/UI/Loading';
+import FetchError from 'components/UI/FetchError';
+
+type DownloadButtonProps = {
+  downloadLink: string;
+};
+const DownloadButton: React.FC<DownloadButtonProps> = ({ downloadLink }) => {
+  const [loading, setLoading] = React.useState<boolean>(false);
+  const [downloadError, setDownloadError] = React.useState<any | null>(null);
+  const downloadFile = () => {
+    setLoading(true);
+    protectedAxios
+      .get<Blob>(downloadLink, {
+        responseType: 'blob',
+      })
+      .then((response: AxiosResponse<Blob>) => {
+        const blob = new Blob([response.data], {
+          type: response.headers['content-type'],
+        });
+        const url = window.URL.createObjectURL(blob);
```

Comment:
Does this need to be `window.URL.createObjectURL`? Or would `URL.createObjectURL` work okay too?

Reply:
I have adjusted this as recommended.